File: Payment_processing.py
# Get the total price of the selected item
# Temporary solution:  should get the order from order input 
total_price = float(input("Enter the total price: $")) 

# Get the total price of the selected item
total_price = float(input("Enter the total price: $"))

# Initialize variables for the customer's payment and change
customer_payment = 0.0

# Function to insert money in cash and coins
def insert_money():
    global customer_payment
    while True:
        try:
            money = float(input("Insert money or coins (enter 0 to finish): $"))
            if money == 0:
                break
            customer_payment += money
        except ValueError:
            print("Invalid input. Please enter a valid amount.")

# Get the customer's payment in cash and coins
print("Insert cash and coins to pay for the item:")
insert_money()

# Calculate the change and provide the receipt
while customer_payment < total_price:
    print("Remaining balance: ${:.2f}".format(total_price - customer_payment))
    print("Insufficient payment. Please provide more cash and coins.")
    insert_money()

change = customer_payment - total_price

print("Thank you for your purchase!")
print(f"Total price: ${total_price:.2f}")
print(f"Customer payment: ${customer_payment:.2f}")
print(f"Change: ${change:.2f}")

File: Payment_processing.cpp
#include <iostream>

// Function to insert money in cash and coins
void insertMoney(float& customerPayment) {
    while (true) {
        try {
            float money;
            std::cout << "Insert money or coins (enter 0 to finish): $";
            std::cin >> money;

            if (money == 0) {
                break;
            }

            customerPayment += money;
        } catch (...) {
            std::cout << "Invalid input. Please enter a valid amount." << std::endl;
            std::cin.clear();
            std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
        }
    }
}

int main() {
    // Get the total price of the selected item
    float totalPrice;
    std::cout << "Enter the total price: $";
    std::cin >> totalPrice;

    // Initialize variables for the customer's payment and change
    float customerPayment = 0.0;

    // Get the customer's payment in cash and coins
    std::cout << "Insert cash and coins to pay for the item:" << std::endl;
    insertMoney(customerPayment);

    // Calculate the change and provide the receipt
    while (customerPayment < totalPrice) {
        std::cout << "Remaining balance: $" << std::fixed << std::setprecision(2) << totalPrice - customerPayment << std::endl;
        std::cout << "Insufficient payment. Please provide more cash and coins." << std::endl;
        insertMoney(customerPayment);
    }

    float change = customerPayment - totalPrice;

    std::cout << "Thank you for your purchase!" << std::endl;
    std::cout << "Total price: $" << std::fixed << std::setprecision(2) << totalPrice << std::endl;
    std::cout << "Customer payment: $" << std::fixed << std::setprecision(2) << customerPayment << std::endl;
    std::cout << "Change: $" << std::fixed << std::setprecision(2) << change << std::endl;

    return 0;
}
